jarviscli/plugins/books.py

```diff
@@ -0,0 +1,78 @@
+from plugin import plugin, require
+from bs4 import BeautifulSoup
+import requests
+
+
+@require(network=True)
+@alias("books", "book")
+@plugin('book_sugges')
+def extract_genre(jarvis, s):
+    """
+        This method gives jarvis the ability to find a list of books
+        based on the user's interest of genres.
+    """
+    jarvis.say("Hey! Ready to turn some pages? \n")
+    jarvis.say("Here are some of the interesting genres: ")
+    genres = [
+        "romance",
+        "fiction",
+        "fantasy",
+        "science-fiction",
+        "non-fiction"
+        "",
+        "children",
+        "history",
+        "mystery",
+        "horror",
+        "historical-fiction",
+        "love",
+        "biography",
+        "thriller",
+        "contemporary",
+        "classics",
+        "novels",
+        "series"]
+    for gen in genres:
+        jarvis.say(gen)
+    jarvis.say("Please let me know your genre: ")
+    genre = jarvis.input()
+    # request sent for the url
+    page = requests.get(
+        "https://www.goodreads.com/search?&query={}".format(genre))
+    soup = BeautifulSoup(page.content, 'html.parser')
+    results = soup.find_all('tr', {'itemtype': "http://schema.org/Book"})
+    records = []
+    jarvis.say("Searching for books......\n")
+
+    for item in results:  # Searches for books
+        try:
+            book = item.find('a', class_='bookTitle').span.text.strip()
+            author = item.find(
+                'div', class_='authorName__container').a.span.text.strip()
+            rating = float(
+                item.find(
+                    'span',
+                    class_="minirating").text.strip().split(" ")[0])
+            record = (book, author, rating)
+        except BaseException:
```

Comment:
Looks good!

I'll take a closer look tomorrow. Just small note, catching all Exceptions is usually not a good idea. Might produce hard-to-find bugs ;).

Reply:
@pnhofmann , I didn't actually catch all exceptions, it was converted automatically by the autopep8 package. It was just except earlier, so as to prevent from entering empty record to the final records list.

Reply:
Same thing if ``except Exceptions`` or just ``except``. Interesting autopep8 prefers the former.

It basically makes finding bugs harder since it hides all Exceptions. Imagine for example, for some reason ratings are measured like 4,5 (with a , like we do it here in Germany for some reason). With your code you will ask yourself, why is no result found instead getting a clean message "ValueError: could not convert string to float: '4,5'.

But please see it just as a general note. In this case the except probably isn't a real issue ;).

Reply:
Yes, I understand, thanks